docs/standard/io/handling-io-errors.md

```diff
@@ -0,0 +1,69 @@
+- <xref:System.NotSupportedException?displayProperty=nameWithType>, which is thrown for invalid colons in .NET Framework.
+- <xref:System.Security.SecurityException?displayProperty=nameWithType>, which is thrown for applications running in limited trust that lack the necessary permissions on .NET Framework only. (Full trust is the default on .NET Framework.)
+
+## Mapping error codes to exceptions
+
+Because the file system is an operating system resource, I/O methods in both .NET Core and .NET Framework wrap calls to the underlying operating system. When an I/O error occurs in code executed by the operating system, the operating system returns error information to the .NET I/O method. The method then translates the error information, typically in the form of an error code, into a .NET exception type. In most cases, it does this by directly translating the error code into its corresponding exception type; it does not perform any special mapping of the error based on the context of the method call.
+
+For example, on the Windows operating system, a method call that returns an error code of `ERROR_FILE_NOT_FOUND` (or 0x2) maps to a <xref:System.IO.FileNotFoundException>, and an error code of `ERROR_PATH_NOT_FOUND` (or 0x03) maps to a <xref:System.IO.DirectoryNotFoundException>.
+
+However, the precise conditions under which the operating system returns particular error codes is often undocumented or poorly documented. As a result, unexpected exceptions can occur. For example, because you are working with a directory rather than a file, you would expect that providing an invalid directory path to the <xref:System.IO.DirectoryInfo.#ctor%6A?displayProperty=nameWithType> constructor throws a <xref:System.IO.DirectoryNotFoundException>. However, it may also throw a <xref:System.IO.FileNotFoundException>.
+
+## Exception handling in I/O operations
+
+Because of this reliance on the operating system, identical exception conditions (such as the directory not found error in our example) can result in an I/O method throwing any one of the entire class of I/O exceptions. This means that, when calling I/O APIs, your code should be prepared to most or all of these exceptions, as shown in the following table:
+
+| Exception type | .NET Core | .NET Framework |
+|---|---|---|
+| <xref:System.IO.IOException> | Yes | Yes |
+| <xref:System.IO.FileNotFoundException> | Yes | Yes |
+| <xref:System.IO.DirectoryNotFoundException> | Yes | Yes |
+| <xref:System.IO.DriveNotFoundException?> | Yes | Yes |
+| <xref:System.IO.PathTooLongException> | Yes | Yes |
+| <xref:System.OperationCancelledException> | Yes | Yes |
+| <xref:System.UnauthorizedAccessException> | Yes | Yes |
+| <xref:System.ArgumentException> | .NET Core 2.0 and earlier| Yes |
+| <xref:System.NotSupportedException> | No | Yes |
+| <xref:System.Security.SecurityException> | No | Limited trust only |
+
+## Handling IOException
```

Comment:
This could be a question for @JeremyKuhne or @jpreese but would exception filters be a good way to manage these different conditions? Is the OS error code stored in the HResult on all platforms?

That would be a good question to answer in this article.

Reply:
Unfortunately we're not in great shape on how we deal with the HResult. For the derived exceptions it is hard coded- so it should always be consistent. For catch-all base class exceptions I don't think we're consistent. Usually it will be the [raw error](https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/Common/src/CoreLib/Interop/Unix/Interop.IOErrors.cs#L170).  There _might_ be places where the PAL in the runtime maps to Windows error codes- I need to spend some time looking.

@pjanotti, Fyi

Reply:
That's good to know. For this article, the right answer may be a note that says the exception type is a much more reliable way to differentiate the error than other data contained in the exception.